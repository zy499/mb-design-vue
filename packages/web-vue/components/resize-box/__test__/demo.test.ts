import demoTest from '../../../scripts/demo-test';

demoTest('resize-box');
