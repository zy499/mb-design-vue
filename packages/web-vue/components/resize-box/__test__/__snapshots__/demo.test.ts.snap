// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<resize-box> demo: render [basic] correctly 1`] = `
"<div>
  <div class=\\"mb-resizebox\\" style=\\"width: 500px; min-width: 100px; max-width: 100%; height: 200px; text-align: center;\\">
    <div class=\\"mb-typography\\">We are building the future of content discovery and creation.
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div role=\\"separator\\" class=\\"mb-divider mb-divider-horizontal\\">
      <!---->
    </div>
    <div class=\\"mb-typography\\"> MaBang's content platforms enable people to enjoy content powered by AI technology. We inform, entertain, and inspire people across language, culture and geography.
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div role=\\"separator\\" class=\\"mb-divider mb-divider-horizontal mb-divider-with-text\\"><span class=\\"mb-divider-text mb-divider-text-center\\">MaBang</span></div>
    <div class=\\"mb-typography\\">Yiming Zhang is the founder and CEO of MaBang.
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-vertical mb-resizebox-direction-right\\">
      <!-- @slot 自定义内容 -->
      <div class=\\"mb-resizebox-trigger-icon-wrapper\\">
        <!-- @slot 自定义 icon --><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-drag-dot-vertical mb-resizebox-trigger-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M17 8h2v2h-2V8ZM17 23h2v2h-2v-2ZM17 38h2v2h-2v-2ZM29 8h2v2h-2V8ZM29 23h2v2h-2v-2ZM29 38h2v2h-2v-2Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path>
          <path d=\\"M17 8h2v2h-2V8ZM17 23h2v2h-2v-2ZM17 38h2v2h-2v-2ZM29 8h2v2h-2V8ZM29 23h2v2h-2v-2ZM29 38h2v2h-2v-2Z\\"></path>
        </svg>
      </div>
    </div>
    <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-horizontal mb-resizebox-direction-bottom\\">
      <!-- @slot 自定义内容 -->
      <div class=\\"mb-resizebox-trigger-icon-wrapper\\">
        <!-- @slot 自定义 icon --><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-drag-dot mb-resizebox-trigger-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M40 17v2h-2v-2h2ZM25 17v2h-2v-2h2ZM10 17v2H8v-2h2ZM40 29v2h-2v-2h2ZM25 29v2h-2v-2h2ZM10 29v2H8v-2h2Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path>
          <path d=\\"M40 17v2h-2v-2h2ZM25 17v2h-2v-2h2ZM10 17v2H8v-2h2ZM40 29v2h-2v-2h2ZM25 29v2h-2v-2h2ZM10 29v2H8v-2h2Z\\"></path>
        </svg>
      </div>
    </div>
  </div>
</div>"
`;

exports[`<resize-box> demo: render [controlled] correctly 1`] = `
"<div>
  <div class=\\"mb-resizebox\\" style=\\"min-width: 100px; max-width: 100%; text-align: center; width: 500px; height: 200px;\\">
    <div class=\\"mb-typography\\">We are building the future of content discovery and creation.
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div role=\\"separator\\" class=\\"mb-divider mb-divider-horizontal\\">
      <!---->
    </div>
    <div class=\\"mb-typography\\"> MaBang's content platforms enable people to enjoy content powered by AI technology. We inform, entertain, and inspire people across language, culture and geography.
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div role=\\"separator\\" class=\\"mb-divider mb-divider-horizontal mb-divider-with-text\\"><span class=\\"mb-divider-text mb-divider-text-center\\">MaBang</span></div>
    <div class=\\"mb-typography\\">Yiming Zhang is the founder and CEO of MaBang.
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-vertical mb-resizebox-direction-right\\">
      <!-- @slot 自定义内容 -->
      <div class=\\"mb-resizebox-trigger-icon-wrapper\\">
        <!-- @slot 自定义 icon --><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-drag-dot-vertical mb-resizebox-trigger-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M17 8h2v2h-2V8ZM17 23h2v2h-2v-2ZM17 38h2v2h-2v-2ZM29 8h2v2h-2V8ZM29 23h2v2h-2v-2ZM29 38h2v2h-2v-2Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path>
          <path d=\\"M17 8h2v2h-2V8ZM17 23h2v2h-2v-2ZM17 38h2v2h-2v-2ZM29 8h2v2h-2V8ZM29 23h2v2h-2v-2ZM29 38h2v2h-2v-2Z\\"></path>
        </svg>
      </div>
    </div>
    <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-horizontal mb-resizebox-direction-bottom\\">
      <!-- @slot 自定义内容 -->
      <div class=\\"mb-resizebox-trigger-icon-wrapper\\">
        <!-- @slot 自定义 icon --><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-drag-dot mb-resizebox-trigger-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M40 17v2h-2v-2h2ZM25 17v2h-2v-2h2ZM10 17v2H8v-2h2ZM40 29v2h-2v-2h2ZM25 29v2h-2v-2h2ZM10 29v2H8v-2h2Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path>
          <path d=\\"M40 17v2h-2v-2h2ZM25 17v2h-2v-2h2ZM10 17v2H8v-2h2ZM40 29v2h-2v-2h2ZM25 29v2h-2v-2h2ZM10 29v2H8v-2h2Z\\"></path>
        </svg>
      </div>
    </div>
  </div>
</div>"
`;

exports[`<resize-box> demo: render [custom-triggers] correctly 1`] = `
"<div class=\\"mb-resizebox\\" style=\\"width: 500px; min-width: 100px; max-width: 100%; height: 200px; text-align: center;\\">
  <div class=\\"mb-typography\\">We are building the future of content discovery and creation.
    <!---->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
  <div role=\\"separator\\" class=\\"mb-divider mb-divider-horizontal\\">
    <!---->
  </div>
  <div class=\\"mb-typography\\"> MaBang's content platforms enable people to enjoy content powered by AI technology. We inform, entertain, and inspire people across language, culture and geography.
    <!---->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
  <div role=\\"separator\\" class=\\"mb-divider mb-divider-horizontal mb-divider-with-text\\"><span class=\\"mb-divider-text mb-divider-text-center\\">MaBang</span></div>
  <div class=\\"mb-typography\\">Yiming Zhang is the founder and CEO of MaBang.
    <!---->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-vertical mb-resizebox-direction-right\\">
    <!-- @slot 自定义内容 -->
    <div class=\\"resizebox-demo resizebox-demo-vertical\\">
      <div class=\\"resizebox-demo-line\\"></div>
    </div>
  </div>
  <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-horizontal mb-resizebox-direction-bottom\\">
    <!-- @slot 自定义内容 -->
    <div class=\\"resizebox-demo resizebox-demo-horizontal\\">
      <div class=\\"resizebox-demo-line\\"></div>
    </div>
  </div>
</div>"
`;

exports[`<resize-box> demo: render [layout] correctly 1`] = `
"<div class=\\"layout-demo\\">
  <section class=\\"mb-layout\\">
    <header class=\\"mb-layout-header\\">Header</header>
    <section class=\\"mb-layout\\">
      <div class=\\"mb-resizebox mb-layout-sider mb-layout-sider-light\\" style=\\"width: 200px;\\">
        <div class=\\"mb-layout-sider-children\\"> Sider </div>
        <!--v-if-->
        <div class=\\"mb-resizebox-trigger mb-resizebox-trigger-vertical mb-resizebox-direction-right\\">
          <!-- @slot 自定义内容 -->
          <div class=\\"mb-resizebox-trigger-icon-wrapper\\">
            <!-- @slot 自定义 icon --><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-drag-dot-vertical mb-resizebox-trigger-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
              <path d=\\"M17 8h2v2h-2V8ZM17 23h2v2h-2v-2ZM17 38h2v2h-2v-2ZM29 8h2v2h-2V8ZM29 23h2v2h-2v-2ZM29 38h2v2h-2v-2Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path>
              <path d=\\"M17 8h2v2h-2V8ZM17 23h2v2h-2v-2ZM17 38h2v2h-2v-2ZM29 8h2v2h-2V8ZM29 23h2v2h-2v-2ZM29 38h2v2h-2v-2Z\\"></path>
            </svg>
          </div>
        </div>
      </div>
      <main class=\\"mb-layout-content\\">Content</main>
    </section>
    <footer class=\\"mb-layout-footer\\">Footer</footer>
  </section>
</div>"
`;
